modules/post/windows/manage/make_token.rb

```diff
@@ -0,0 +1,94 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Post
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Make Token Command',
+        'Description' => %q{
+          In its default configuration, this module creates a new network security context with the specified 
+          logon data (username, domain and password). Under the hood, Meterpreter's access token is cloned, and
+          a new logon session is created and linked to that token. The token is then impersonated to acquire 
+          the new network security context. This module has no effect on local actions - only on remote ones 
+          (where the specified credential material will be used). This module does not validate the credentials 
+          specified.
+        },
+        'License' => MSF_LICENSE,
+        'Notes' => {
+          'AKA' => ['make_token', 'maketoken']
+        },
+        'Platform' => ['win'],
+        'SessionTypes' => ['meterpreter'],
+        'Author' => [
+          'Daniel López Jiménez (attl4s)', 
+          'Simone Salucci (saim1z)'
+        ],
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_railgun_api
+              stdapi_sys_config_revert_to_self
+              stdapi_sys_config_update_token
+            ]
+          }
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('DOMAIN', [true, 'Domain to use' ]),
+        OptString.new('USER', [true, 'Username to use' ]),
+        OptString.new('PASSWORD', [true, 'Password to use' ])
+      ]
+    )
+
+    register_advanced_options(
+      [
+        OptString.new('LOGONTYPE', [true, 'The type of logon operation to perform. Using LOGON32_LOGON_INTERACTIVE may cause issues within the session (typically due to the token filtering done by the UserAccountControl mechanism in Windows). Use with caution', 'LOGON32_LOGON_NEW_CREDENTIALS'])
```

Comment:
Per the docs this can be one of the following, not a combination. For that reason, this should probably be an `OptEnum` so the user knows what values are available and is limited to those values.

https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-logonusera

That'd make it something like

```ruby
OptEnum.new('LOGONTYPE', [true, 'The type of logon operation to perform. Using LOGON32_LOGON_INTERACTIVE may cause issues within the session (typically due to the token filtering done by the UserAccountControl mechanism in Windows). Use with caution', 'LOGON32_LOGON_NEW_CREDENTIALS', %w[ LOGON32_LOGON_BATCH LOGON32_LOGON_INTERACTIVE LOGON32_LOGON_NETWORK ...] ])

Reply:
Changed in https://github.com/rapid7/metasploit-framework/pull/18022/commits/623657b0721a1bca1fb9259851f2b908e2657c2d